main.go

```diff
@@ -29,36 +28,48 @@ import (
 */
 
 func main() {
-	//term := os.Args[1]
-
-	var wg sync.WaitGroup
-	var mutex sync.Mutex
+	term := os.Args[1]
 
 	files, e := os.ReadDir("./testfiles")
 	if e != nil {
 		log.Fatal(e)
 	}
 
+	contents2 := make(map[string][][]byte, len(files))
+
 	for _, f := range files {
-		wg.Add(1)
-		go func(fi fs.DirEntry) {
-			defer wg.Done()
-			mutex.Lock()
-			defer mutex.Unlock()
-
-			/*
-				The code below would require a mutex because of the multiple calls to fmt.Println.
-
-				fmt.Println(" __,  _, _,_ _, _ __,")
-				fmt.Println(" |_  / \\ | | |\\ | | \\")
-				fmt.Println(" |   \\ / | | | \\| |_/")
-				fmt.Println(" ~    ~  `~' ~  ~ ~  ")
-				fmt.Println("")
-
-			*/
-			fmt.Println(fi.Name())
-		}(f) // always explicitly pass this argument if looping + generating goroutines
+		content, e := os.ReadFile(fmt.Sprintf("./testfiles/%s", f.Name()))
+		if e != nil {
+			log.Fatal(e)
+		}
+
+		contents2[f.Name()] = split(content, 10)
+	}
+
+	var hits []string
+
+	for fname, lines := range contents2 {
+		for i, line := range lines {
+			if strings.Contains(string(line), term) {
+				hits = append(hits, fmt.Sprintf("%s:%d", fname, i+1))
+			}
+		}
+	}
+
+	fmt.Println(hits)
+}
+
+func split(input []byte, delimiter byte) [][]byte {
+	var container []byte
+	var result [][]byte
+	for _, b := range input {
+		if b == delimiter {
+			result = append(result, container)
+			container = make([]byte, 0)
```

Comment:
Instead of using `make([]byte, 0)` to create an empty byte slice, you can simply use an empty byte slice literal `[]byte{}`.